graphql_compiler/interpreter/expression_ops/immediate_expression_handlers.py

```diff
@@ -41,6 +41,9 @@ def evaluate_context_field(
     location = expression.location.at_vertex()
     field_name = expression.location.field
 
+    # TODO(bojanserafimov): Memoize hints for each location.
+    hints = construct_hints_for_location(query_metadata_table, query_arguments, location)
```

Comment:
Can you add a similar thing around line 29?

In the output-generation stage, `project_property` calls are made while resolving `ContextField` expressions so this works for the particular test case. However, when filtering, `project_property` calls come from `LocalField` expressions (because the property is in the local scope) so this code path won't get used. Even more confusingly for the end user, if using a tagged parameter for a filter, the `project_property` for the tagged value will have hints because it too will be a `ContextField`.